salt/crypt.py

```diff
@@ -593,64 +593,71 @@ def _authenticate(self):
                                                                 crypt='clear',
                                                                 io_loop=self.io_loop)
-        error = None
-        while True:
-            try:
-                creds = yield self.sign_in(channel=channel)
-            except SaltClientError as exc:
-                error = exc
-                break
-            if creds == 'retry':
-                if self.opts.get('detect_mode') is True:
-                    error = SaltClientError('Detect mode is on')
+        try:
+            error = None
+            while True:
+                try:
+                    creds = yield self.sign_in(channel=channel)
+                except SaltClientError as exc:
+                    error = exc
                     break
-                if self.opts.get('caller'):
-                    # We have a list of masters, so we should break
-                    # and try the next one in the list.
-                    if self.opts.get('local_masters', None):
-                        error = SaltClientError('Minion failed to authenticate'
-                                                ' with the master, has the '
-                                                'minion key been accepted?')
+                if creds == 'retry':
+                    if self.opts.get('detect_mode') is True:
+                        error = SaltClientError('Detect mode is on')
                         break
-                    else:
-                        print('Minion failed to authenticate with the master, '
-                              'has the minion key been accepted?')
-                        sys.exit(2)
-                if acceptance_wait_time:
-                    log.info(
-                        'Waiting %s seconds before retry.', acceptance_wait_time
-                    )
-                    yield tornado.gen.sleep(acceptance_wait_time)
-                if acceptance_wait_time < acceptance_wait_time_max:
-                    acceptance_wait_time += acceptance_wait_time
-                    log.debug(
-                        'Authentication wait time is %s', acceptance_wait_time
+                    if self.opts.get('caller'):
+                        # We have a list of masters, so we should break
+                        # and try the next one in the list.
+                        if self.opts.get('local_masters', None):
+                            error = SaltClientError('Minion failed to authenticate'
+                                                    ' with the master, has the '
+                                                    'minion key been accepted?')
+                            break
+                        else:
+                            print('Minion failed to authenticate with the master, '
+                                  'has the minion key been accepted?')
+                            sys.exit(2)
```

Comment:
Will this cause the minion process to stop trying to connect with the master?

Reply:
I'm not changing behavior. It just makes sure we close the channel when done.
I.e., whatever was the existing reconnection behavior, is still in place.